programs/pkey/key_app.c

```diff
@@ -181,7 +183,8 @@ int main( int argc, char *argv[] )
         mbedtls_printf( "\n  . Loading the private key ..." );
         fflush( stdout );
 
-        ret = mbedtls_pk_parse_keyfile( &pk, opt.filename, opt.password );
+        ret = mbedtls_pk_parse_keyfile( &pk, opt.filename, opt.password,
+                                        mbedtls_test_rnd_std_rand, NULL );
```

Comment:
`mbedtls_test_rnd_std_rand` is for testing only and is not suitable for sample programs. Sample programs that require a RNG must depend on the entropy module and a DRBG module.

Reply:
Agreed, that's also noted in the PR description. However I was thinking perhaps we could leave this for later. We're not supposed to use `MBEDTLS_PRIVATE()` in sample programs either, but we decided to live with it until we fix it in 3.x.

(This program is rather easy to fix, but `ssl_server2` is a bit more complex, as we need a RNG for reading private key while parsing the `sni` option, but we want to wait until we parse all command-line options before we seed the RNG because there's a `reproducible` option... Some refactoring would be in order.)

Reply:
>  but `ssl_server2` is a bit more complex

Actually it isn't, I was confused. The RNG is actually seeded early enough. So I'm fixing this as well.